api2-war/src/main/java/eu/europeana/api2/v2/web/controller/SearchController.java

```diff
@@ -85,643 +82,1074 @@
-            response.breadCrumbs = NavigationUtils.createBreadCrumbList(query);
-        }
-        if (StringUtils.containsIgnoreCase(profile, "spelling") || StringUtils.containsIgnoreCase(profile, "portal")) {
-            response.spellcheck = ModelUtils.convertSpellCheck(resultSet.getSpellcheck());
-        }
-        // if (StringUtils.containsIgnoreCase(profile, "suggestions") ||
-        // StringUtils.containsIgnoreCase(profile, "portal")) {
-        // }
-        return response;
-    }
-
-    @RequestMapping(value = "/v2/search.kml", produces = {"application/vnd.google-earth.kml+xml", MediaType.ALL_VALUE})
-    // @RequestMapping(value = "/v2/search.kml", produces =
-    // "application/vnd.google-earth.kml+xml")
-    public
-    @ResponseBody
-    KmlResponse searchKml(
-            @RequestParam(value = "query", required = true) String queryString,
-            @RequestParam(value = "qf", required = false) String[] refinements,
-            @RequestParam(value = "start", required = false, defaultValue = "1") int start,
-            @RequestParam(value = "wskey", required = true) String wskey,
-            HttpServletRequest request,
-            HttpServletResponse response) throws Exception {
-
-        // workaround of a Spring issue
-        // (https://jira.springsource.org/browse/SPR-7963)
-        String[] _qf = request.getParameterMap().get("qf");
-        if (_qf != null && _qf.length != refinements.length) {
-            refinements = _qf;
-        }
+    if (null != video_hd) {
+      videoHDs.add(video_hd);
+    }
 
-        try {
-            ApiKey apiKey = apiService.findByID(wskey);
-            apiService.checkReachedLimit(apiKey);
-        } catch (DatabaseException e) {
-            response.setStatus(401);
-            throw new Exception(e);
-        } catch (LimitReachedException e) {
-            response.setStatus(429);
-            throw new Exception(e);
-        }
-        KmlResponse kmlResponse = new KmlResponse();
-        Query query = new Query(SearchUtils.rewriteQueryFields(queryString))
-                .setRefinements(refinements)
-                .setApiQuery(true)
-                .setAllowSpellcheck(false)
-                .setAllowFacets(false);
-        query.setRefinements("pl_wgs84_pos_lat_long:[* TO *]");
-        try {
-            ResultSet<BriefBean> resultSet = searchService.search(BriefBean.class, query);
-            kmlResponse.document.extendedData.totalResults.value = Long.toString(resultSet.getResultSize());
-            kmlResponse.document.extendedData.startIndex.value = Integer.toString(start);
-            kmlResponse.setItems(resultSet.getResults());
-            // Disabled while awaiting better implementation (ticket #1742)
-            // apiLogService.logApiRequest(wskey, query.getQuery(), RecordType.SEARCH_KML, "kml");
-        } catch (SolrTypeException e) {
-            response.setStatus(429);
-            throw new Exception(e);
+    if (null != image_aspectratio) {
```

Comment:
thanks. refactored them as you suggested. it's best to be aligned with the style of the other ones.